server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1570,7 +1572,11 @@ else if (!poolQuantities.keySet().equals(entitlements.keySet())) {
         consumerCurator.update(consumer);
 
         handler.handlePostEntitlement(this, consumer, entitlements);
-        handler.handleSelfCertificates(consumer, poolQuantities, entitlements);
+
+        // Distributors and shares don't need entitlement certificate since they don't talk to the CDN
+        if (!isDistributor && !isShare) {
```

Comment:
The distributor certs must be generated. These certs are what allows the satellite to sync content from the upstream CDN into the satellite.